Cargo.toml

```diff
@@ -82,3 +82,6 @@ opt-level = 3 # bs58 library is too slow to use in debug
 expensive_tests = []
 regression_tests = []
 old_tests = []
+
+[patch.crates-io]
+rocksdb = { git = "https://github.com/nearprotocol/rust-rocksdb", branch = "lifetime" }
```

Comment:
Without adding the lifetime to iterator_cf_opt, rust complains you can't reference db coz it's owned by current function. With @frol's help it turns out to be iterator_cf_opt doesn't give proper lifetime marker on definition, We had a discussion in #dev last Fri.

Reply:
That's not okay, we did not get rid of Parity fork to replace it with our fork.

Reply:
So my complain about `<'a, 'b:'a>` actually referred to the change that you have introduced: https://github.com/nearprotocol/rust-rocksdb/commit/c28eb1d705f73c4a30c1dd0b11b4d4b9c330139b . This change is incorrect, iterator cannot live longer then the stuff it is iterating over. The only reason the compiler allows it is because it `DBRawIterator` works with raw pointers.

Reply:
Yup. Do not have a better ideas yet, so reported as an issue to see if they can do the proper fix https://github.com/rust-rocksdb/rust-rocksdb/issues/356

Reply:
I fixed it, it should work now. We can get rid of our rust-rocksdb fork. I suspect the reason it did not work is because we were using `RwLock` before which makes the lifetimes tricky.